triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -671,30 +713,32 @@ private[lf] class Runner private (
       }
 
       UnfoldState(v)({ state =>
-        numberOfRuleEvaluations += 1
-
         go(state) match {
           case next @ -\/(_) =>
-            val endTime = System.nanoTime()
-            triggerContext.logInfo(
-              "Trigger rule evaluation end",
-              "metrics" -> LoggingValue.Nested(
-                LoggingEntries(
-                  "steps" -> numberOfRuleEvaluations,
-                  "submissions" -> LoggingValue.Nested(
-                    LoggingEntries(
-                      "total" -> numberOfSubmissions,
-                      "create" -> numberOfCreates,
-                      "exercise" -> numberOfExercises,
-                      "createAndExercise" -> numberOfCreateAndExercise,
-                      "exerciseByKey" -> numberOfExerciseByKey,
-                    )
-                  ),
-                  "duration" -> s"${FiniteDuration(endTime - startTime, "nanos").toMillis}ms",
-                )
-              ),
-            )
-            next
+            try {
+              val endTime = System.nanoTime()
+              triggerContext.logInfo(
+                "Trigger rule evaluation end",
+                "metrics" -> LoggingValue.Nested(
+                  LoggingEntries(
+                    "steps" -> numberOfRuleEvaluations,
+                    "submissions" -> LoggingValue.Nested(
+                      LoggingEntries(
+                        "total" -> numberOfSubmissions,
+                        "create" -> numberOfCreates,
+                        "exercise" -> numberOfExercises,
+                        "createAndExercise" -> numberOfCreateAndExercise,
+                        "exerciseByKey" -> numberOfExerciseByKey,
+                      )
+                    ),
+                    "duration" -> s"${FiniteDuration(endTime - startTime, "nanos").toMillis}ms",
+                  )
+                ),
+              )
+              next
+            } finally {
+              discard(ruleEvaluationTimer.cancel())
```

Comment:
Using `finally` as we need to ensure the timer is stopped (regardless of whether the try block succeeds or throws). If the timer can not be cancelled, then something bad is happening(?) and the trigger will be stopped (because the exception will be thrown).